File: environment-spring-boot/casadocodigo/src/main/java/org/ecorp/casadocodigo/services/PaisService.java
package org.ecorp.casadocodigo.services;

import java.util.Optional;
import javax.validation.Valid;
import org.ecorp.casadocodigo.dtos.PaisDTO;
import org.ecorp.casadocodigo.exceptions.ResourceNotFoundException;
import org.ecorp.casadocodigo.forms.PaisFormRequest;
import org.ecorp.casadocodigo.model.Pais;
import org.ecorp.casadocodigo.repositories.PaisRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class PaisService {

  @Autowired
  private PaisRepository repository;

  public PaisDTO criar(@Valid PaisFormRequest paisRequest) {
    Pais pais = repository.save(paisRequest.toModel());
    return new PaisDTO(pais);
  }

  public PaisDTO buscaPorId(Long id) {
    Optional<Pais> found = repository.findById(id);
    if (found.isEmpty())
      throw new ResourceNotFoundException();
    return new PaisDTO(found.get());
  }

}
//

File: src/main/kotlin/com/tiozao/cdd/loja/domain/service/PaisService.kt
package com.tiozao.cdd.loja.domain.service

import com.tiozao.cdd.loja.domain.model.EstadoModel
import com.tiozao.cdd.loja.domain.model.PaisModel
import com.tiozao.cdd.loja.repository.EstadoRepository
import com.tiozao.cdd.loja.repository.PaisRepository
import com.tiozao.cdd.loja.repository.extensions.toEntity
import com.tiozao.cdd.loja.repository.extensions.toModel
import org.springframework.stereotype.Service
import org.springframework.validation.annotation.Validated
import javax.validation.Valid

@Service
@Validated
class PaisService(
    private var paisRepository: PaisRepository,
    private var estadoRepository: EstadoRepository
    ) {

    fun createPais(@Valid paisModel: PaisModel): PaisModel {
        return paisRepository.save(paisModel.toEntity())
            .toModel()
    }

    fun addEstadoToPais(@Valid estadoModel: EstadoModel, paisNome: String): EstadoModel {
        var pais = paisRepository.findByNome(paisNome).get()
        return estadoRepository.save(estadoModel
            .toEntity().copy(pais = pais))
            .toModel()
    }


}